topics/materials-science/tutorials/muspinsim/tutorial.md

```diff
@@ -0,0 +1,932 @@
+>    - *"Plot Title"*: `Hyperfine (with powder averaging) Example`
+>    - *"Label for x axis"*: `B`
+>    - *"Label for y axis"*: `Asymmetry`
+>    - *"Output File Type"*: `PNG`
+>    - *"Muspinsim File Series"*:
+>      - *"Muspinsim File Series 1"*:
+>        - {% icon param-file %} *"Muspinsim Experiment Data (.dat)"*: `alc.dat`
+>        - *"Line/Point Color"*: `Black`
+>        - *"Series Type"*: `Line`
+>        - *"Line type"*: `Solid`
+>        - *"Line width"*: `1.0`
+>      - *"Muspinsim File Series 2"*:
+>        - {% icon param-file %} *"Muspinsim Experiment Data (.dat)"*: `alc.dat`
+>        - *"Line/Point Color"*: `Blue`
+>        - *"Series Type"*: `Points`
+>        - *"Point Type"*: `Square`
+>        - *"Point Scale"*: `2.0`
+> 2. Should generate the following output(s):
+>    - {% icon param-file %} `MuSpinSim Plot on data 4` (Output dataset)
+>
+{: .hands_on}
+
+![Resultant plot of asymmetry against magnetic field for the ALC example](../../images/alc_example.png "The plot shows two distinct peaks in response to the magnetic field")
+
+# Workflow 5 - Temperature
+
+In this example we look at the effect of temperature on the system. Many of the
+parameters are the same as they were in
+[Workflow 2](#workflow-2---hyperfine-coupling), but with an external static
+field of $$1T$$. This means that the difference in energy between the "up" and
+"down" states for the electron is given by:
+
+$$
+\Delta E = 2\pi\hbar \gamma_e \cdot (1 \,\mathrm{T}) =  2\pi\hbar \cdot (28024.95 \,\mathrm{MHz}) \approx 1.159\cdot 10^{-4} \,\mathrm{eV}
+$$
+
+The scale of this energy is much higher than anything that can come out of the
+hyperfine coupling between electron and muon (only 10 MHz), so we're safely in
+the area of validity of the high field approximation. In terms of temperature
+scale, we expect the high temperature ($$T \approx \infty$$) regime will be
+valid for:
+
+$$
+T \gg \frac{1.159\cdot 10^{-4} \,\mathrm{eV}}{ k_B} \approx 1.35 \,\mathrm{K}.
+$$
+
+For this reason, we make a comparison between two versions of the same system, one with $$T=\infty$$ and one with $$T = 1 \,\mathrm{K}$$, low enough to be out of the high temperature regime. The expected behaviour is as follows:
+
+* the external field sets a base Larmor frequency of 135.54 MHz for the muon. The hyperfine interaction adds or subtracts 5 MHz based on whether the electron is up or down;
+* for very low T, the electron will start entirely in one state, aligned with the external magnetic field. In this case we expect the signal to be a single cosinusoid at 140.54 MHz;
+* for very high T, the electron will be in a statistical ensemble of up and down state, and we expect the signal to be an average of two cosinusoids at 140.54 and 130.54 MHz respectively.
+
+## MuSpinSim Configure
+
+As we are simulating two different temperatures, we need a configuration file
+for each.
```

Comment:
wrt this and the above comment, I only learned about the concept of `_file_ranges` in MSS after writing this (agree it's a very natural point to introduce it), and the original workflow had two inputs so assumed it **had** to be that way. I'll experiment with the workflow and see if there's a way of extracting the two files from the dataset automatically (obviously possible manually as you describe).

Reply:
OK so I haven't been able to get exactly the behaviour we want (in a workflow context). I can wire it up like I would any other, with two temps in the config and providing a collection to the plot:

![image](https://user-images.githubusercontent.com/61705287/211831897-3ba309c9-89f5-41b6-9ce2-edf7302e6817.png)

The collection output from contains two datasets, but I can see no way of "splitting" them and feeding dataset 0 into the plotting tool to show as a line, and dataset 1 to have points. Running like this generates two distinct graphs that are using the same exact settings to generate it:

![image](https://user-images.githubusercontent.com/61705287/211832650-6971cd9d-2f00-4162-bb2e-43b1ec3ccbd6.png)
![image](https://user-images.githubusercontent.com/61705287/211832697-7895a0c6-7cd2-4892-b4c2-bf6b967b04ec.png)

So it's a choice between having two configs/simulations but one plot, or one config/simulation but we generate two graphs of low/high T which cannot be labeled/coloured differently. Personally I think keeping things as they are is better, rather than not being able to associate each graph with it's experimental conditions. What do you think @elichad ?

Reply:
A third option - there are some 'Collection Operations' tools available on usegalaxy.eu which can be used to extract datasets from collections (`Extract dataset` specifically - not sure if there's a tool out there that just unpacks every dataset from a collection). For example, this workflow creates the desired plot (the two `Extract dataset` steps are set to different indexes):
![Workflow using Extract Dataset tool to unpack the collection before plotting](https://user-images.githubusercontent.com/20683271/211876426-5f201e2d-9869-4e48-92db-081ea6c09169.png)
We can enable these tools on Muon Galaxy if needed. Don't know if it's actually a better option though.